### ISSUE TITLE
Add unit tests for the crawler

### ISSUE THREAD

**Issue by marco-c** [marco-c/coverage-crawler#32]
So far, we have no tests.

**Comment by Sush0512** [comment]
Yes, I would like to work on it. Shall we create a `test folder` and add all the unit tests?

**Comment by marco-c** [comment]
Yes! Let's start by changing the project structure to put a test folder with an empty test and a directory "coverage_crawler" where we put the code.
We can use pytest for testing.

**Comment by marco-c** [comment]
That sounds good, I would file a separate issue for changing the project structure.

**Comment by NNRepos** [comment]
@marco-c This repo looks pretty inactive, but if you're still here, could you please suggest some tests we could implement?

**Comment by marco-c** [comment]
@NNRepos it is inactive at the moment, but the plan is to work on this when time will allow.
Regarding tests, we have basically 0 coverage for the crawler, so you could start by adding tests for the simplest functions in crawler.py.

**Comment by nnrepos** [comment]
hey @marco-c, i created a unittest to see if i understand what you need. please let me know what you think 😄

**Comment by nnrepos** [comment]
thanks for the support @marco-c :)
i would like to know - what should the next steps be? would you like me to add a unittest for every crawler method? perhaps some integration tests? something specific you would like to be tested?
a task list would be appreciated

**Comment by marco-c** [comment]
@nnrepos we should add a unit test for each function in the crawler (when it is possible, of course). Feel free to open a PR for each test/function.

**Comment by nnrepos** [comment]
@marco-c alright.
since we don't want to create calls to `google.com`, is there any other website i am allowed to test on? something that is expected to rarely (hopefully never) change, and is reliable? perhaps the page of this repository in github?

**Comment by marco-c** [comment]
@nnrepos ideally we should run a local server so we can fully control the contents.
